yaml/src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLParser.java

```diff
@@ -1012,6 +1012,10 @@ public Object getNumberValueDeferred() throws IOException {
         // 01-Feb-2023, tatu: ParserBase implementation does not quite work
         //   due to refactoring. So let's try to cobble something together
 
+        if (_cleanedTextValue == null) {
+            _reportError("Current token ("+_currToken+") not numeric, can not use numeric value accessors");
```

Comment:
Huh? Based on what?

Let's start with basics: first a failing unit test, then see how to fix it. I think you have identified a problem, correctly, but proposed is probably masking the root cause.

Reply:
https://github.com/FasterXML/jackson-dataformats-text/pull/446/commits/30414def0dab0fd2bae79d8c16f7501220801e09